pkg/controller/controller_utils.go

```diff
@@ -466,11 +467,7 @@ func getPodsPrefix(controllerName string) string {
 	return prefix
 }
 
-func (r RealPodControl) CreatePods(namespace string, template *v1.PodTemplateSpec, object runtime.Object) error {
-	return r.createPods("", namespace, template, object, nil)
-}
-
-func (r RealPodControl) CreatePodsWithControllerRef(namespace string, template *v1.PodTemplateSpec, controllerObject runtime.Object, controllerRef *metav1.OwnerReference) error {
+func validateControllerRef(controllerRef *metav1.OwnerReference) error {
```

Comment:
This validation is only required for controllers. API server doesn't know if a request is sent by a controller thus cannot do the validation.

If a TPR creates dependents, then its custom controller could do a similar check.